tests/test_sns/test_subscriptions_boto3.py

```diff
@@ -477,9 +481,136 @@ def test_subscribe_invalid_filter_policy():
             Endpoint="http://example.com/",
             Attributes={"FilterPolicy": json.dumps({"store": [1000000001]})},
         )
+        assert False
     except ClientError as err:
         err.response["Error"]["Code"].should.equal("InternalFailure")
 
+    try:
+        conn.subscribe(
+            TopicArn=topic_arn,
+            Protocol="http",
+            Endpoint="http://example.com/",
+            Attributes={
+                "FilterPolicy": json.dumps({"price": [{"numeric": ["<", "100"]}]})
+            },
+        )
+        assert False
+    except ClientError as err:
+        err.response["Error"]["Code"].should.equal("InvalidParameter")
+        err.response["Error"]["Message"].should.equal(
+            "Invalid parameter: Attributes Reason: FilterPolicy: Value of < must be numeric\n at ..."
+        )
```

Comment:
```suggestion
    with pytest.raises(ClientError) as err:
        conn.subscribe(
            TopicArn=topic_arn,
            Protocol="http",
            Endpoint="http://example.com/",
            Attributes={
                "FilterPolicy": json.dumps({"price": [{"numeric": ["<", "100"]}]})
            },
        )

    err.response["Error"]["Code"].should.equal("InvalidParameter")
    err.response["Error"]["Message"].should.equal(
        "Invalid parameter: Attributes Reason: FilterPolicy: Value of < must be numeric\n at ..."
    )
```
This is better readable, IMO. This way we don't need the `assert False` - pytest will automatically fail if the expected error isn't raised.

(Just to be clear, you don't have to change the existing tests in this PR! As long as the new tests follow the correct approach, we can always fix the others later.)

Reply:
Well, that's what you get for writing tests at night :D I completely missed that you can actually get the exception instance thrown with `pytest.raises()`. Fixing this up